components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImplTest.java

```diff
@@ -149,6 +149,24 @@ public void testGetClaimsFilteredByOIDCScopes(String requestedScope, int numberO
         Assert.assertEquals(claims.get(0), claim);
     }
 
+    @Test
+    public void testHandleUpdatedAtClaim() {
+
+        String date = "2021-10-27T11:34:13.791Z";
+        claims = new HashMap<>();
+        claims.put("updated_at", date);
+        String[] requestedScopes = {"profile"};
+        OIDCScopeClaimCacheEntry oidcScopeClaimCacheEntry = new OIDCScopeClaimCacheEntry();
+        oidcScopeClaimCacheEntry.setScopeClaimMapping(getScopeDTOList());
+        OIDCScopeClaimCache.getInstance().addScopeClaimMap(-1234, oidcScopeClaimCacheEntry);
+        Map<String, Object> filteredClaims = openIDConnectClaimFilter.getClaimsFilteredByOIDCScopes(claims,
+                requestedScopes, CLIENT_ID, SP_TENANT_DOMAIN);
+        // Due to the effect of time zone during time conversion, considering only the seconds during comparison
+        // Check is to ensure this claim is in seconds not milliseconds
+        String filteredDate = String.valueOf(filteredClaims.get("updated_at"));
+        Assert.assertEquals(filteredDate.substring(8), "53");
```

Comment:
Shall we do a full string match here?

Reply:
The date time conversion in the method outputs the local date time. Therefore hardcoding the the converted value in the test will fail the test in different timezones. Only the seconds are compared here to get rid of this time zone difference.